### PR TITLE
Bug fixes from PE-D

### LINKED ISSUE #169: [PE-D][Tester A] Policy expiry date should not be allowed to be earlier than policy issue date
As shown in the screenshot, the user is allowed to enter a policy expiry date earlier than the issue date, which should not be allowed.

This could be detrimental if lets say a persons policy is issued on a certain date but in the person it is already marked as expired.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: AaronJT1/ped#6

### LINKED ISSUE #177: [PE-D][Tester A] Editing person with no policy should show proper error message
Since David in the screenshot does not have policy, there should be a message which tells the user so. The message in this case is to provide edits for all policy parameters, which the user already did. 

This could be detrimental if the database contains hundreds of users and the user spends a long time trying to figure out which policy parameter he did not provide edits when in fact the person he is trying to change does not have a policy in the first place

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: AaronJT1/ped#9

### LINKED ISSUE #186: [PE-D][Tester B] Traces of AB3 are still left in the codebase
Traces of the AB3 references are still left in the code base. Upon entering the `clear` command to clear all entries within Insureiq, the output still states "Address book has been cleared!". It would be nice to update it such that it displays "Insureiq has been cleared!"

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: SelwynAng/ped#10

### PR DESCRIPTION
**Pull request by RSXIX**
Bug fixes for the following issues

1. Throw error when policy issue date comes is not before policy expiration date. Closes #169 
2. Comprehensive error message including missing parameters when editing a person with default policy and not providing all parameters. Closes #177 
3. Replaced "Address Book" messages with "Client list" messages. Closes #186

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_SUCCESS = "Client has been cleared!";
```
Comment (brendanneojw) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java]: Minor typo here, forget to include the word `list`

```suggestion
    public static final String MESSAGE_SUCCESS = "Client list has been cleared!";
```

Comment (KumChaiYin) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java]:
```suggestion
    public static final String MESSAGE_SUCCESS = "Client list has been cleared!";
```

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -127,6 +128,10 @@ public AddCommand parse(String args) throws ParseException {
             policyExpiryDate = ParserUtil.parsePolicyExpiryDate(argMultimap.getValue(PREFIX_POLICY_EXPIRY_DATE).get());
         }
 
+        if (policyIssueDate.compareTo(policyExpiryDate) > 0) {
+            throw new ParseException(MESSAGE_DATES_NOT_COMPATIBLE);
+        }
```
Comment (KumChaiYin) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:133]: Nice in enforcing the rule that expiry date should come after issue date

File: src/main/java/seedu/address/model/policy/Policy.java [hunk 3]
```diff
@@ -73,6 +73,75 @@ public String toString() {
...
+     * @return int
+     */
+    public int compareDates() {
+        return this.policyExpiryDate.compareTo(policyIssueDate);
+    }
+
+    /**
+     * Returns true if policy is same as default
+     */
+    public boolean isDefaultPolicyProfile() {
```
Comment (KumChaiYin) [inline src/main/java/seedu/address/model/policy/Policy.java:89]: it is a very useful method! Nice!

File: src/main/java/seedu/address/model/policy/Policy.java [hunk 4]
```diff
@@ -73,6 +73,75 @@ public String toString() {
...
     }
 
+    /**
+     * Compares the policy's issue date and expiry date
+     * Returns more than 1 if the expiry date is after issue date
+     * Returns less than 1 otherwise
+     * @return int
+     */
+    public int compareDates() {
+        return this.policyExpiryDate.compareTo(policyIssueDate);
```
Comment (brendanneojw) [inline src/main/java/seedu/address/model/policy/Policy.java:83]: Good way to compare the 2 dates

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -112,31 +118,49 @@ public boolean isSamePerson(Person otherPerson) {
...
 
     /**
      * Returns true if the specified person has any policy parameters equal to the default
      */
     public boolean hasAnyDefaultPolicyParameters() {
-        return this.getPolicy().getPolicyNumber().value.equals(PolicyNumber.DEFAULT_VALUE)
-                || this.getPolicy().getPolicyExpiryDate().date.toString().equals(PolicyDate.DEFAULT_COMPARISON_VALUE)
-                || this.getPolicy().getPolicyIssueDate().date.toString().equals(PolicyDate.DEFAULT_COMPARISON_VALUE)
-                || this.getPolicy().getCompany().value.equals(Company.DEFAULT_VALUE);
+        return this.policy.hasDefaultParameter();
```
Comment (KumChaiYin) [inline src/main/java/seedu/address/model/person/Person.java:128]: Excellent in implementing the principle "Tell, don't ask"

**Review by brendanneojw — COMMENTED** [review]
Overall LGTM!

**Review by KumChaiYin — APPROVED** [review]
LGTM

**Pull request merged**